### PR TITLE
raft_serverpb: add the extra trait `use_plain_file` for SnapshotMeta

### PR DESCRIPTION
**Pull request by LykxSassinator**
## Description
This PR is a pre-requisite for TiKV #https://github.com/tikv/tikv/pull/17408/files, used for adding a new trait `use_plain_file` for SnapshotMeta. And this trait is introduced for marking that the  relative snapshot can be flushed with `plain file` or `sst files`.

### CODE REVIEW

File: pkg/configpb/configpb.pb.gw.go [hunk 1]
```diff
@@ -31,7 +30,6 @@ var _ status.Status
 var _ = runtime.String
 var _ = utilities.NewDoubleArray
 var _ = descriptor.ForMessage
-var _ = metadata.Join
```
Comment (overvenus) [inline pkg/configpb/configpb.pb.gw.go:34]: Why is this file changed?

File: proto/raft_serverpb.proto [hunk 2]
```diff
@@ -59,6 +59,9 @@ message SnapshotMeta {
     // It should only be used for v2 to send snapshot to v1.
     // See https://github.com/pingcap/tiflash/issues/7568
     uint64 commit_index_hint = 7;
+    // Flag reflects whether the snapshot is persisted with plain files
+    // or sst files. `true` means using plain files.
+    bool use_plain_file = 8;
```
Comment (overvenus) [inline proto/raft_serverpb.proto:64]: How about adding it to `SnapshotCFFile`? It's more flexible, eg. index regions have little data in the default CF but a large amount of data in the write CF.

File: pkg/configpb/configpb.pb.gw.go (hunk 1 above)
Comment (LykxSassinator) [reply pkg/configpb/configpb.pb.gw.go:34]: It's cleaned by script automatically.

File: proto/raft_serverpb.proto (hunk 2 above)
Comment (LykxSassinator) [reply proto/raft_serverpb.proto:64]: Reasonable. And the TODO -- "dump the plain file based on the size of each column family respectively" can be implemented based on it.

**Comment by LykxSassinator** [comment]
To ensure the forward compatibility of TiKV, this change can be replaced with another implementation, by doing apply check on the `recv` side of TiKV.
So, this change is not necessary and can be safely closed.
